pkg/controller/config/config_controller_test.go

```diff
@@ -0,0 +1,57 @@
+package config
+
+import (
+	"testing"
+
+	mf "github.com/jcrossley3/manifestival"
+	op "github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TestTransformManifest_WithAnnotation(t *testing.T) {
+	resourceWithAnnotation := "../../../deploy/resources/v0.9.2/addons/pipelines/s2i-go/s2i-go-build-deploy.yaml"
+	assert := func(u unstructured.Unstructured, expected string) {
+		v, _, _ := unstructured.NestedMap(u.Object, "metadata")
+		ns := v["namespace"]
+		if ns != expected {
+			t.Errorf("Expected '%s', got '%s'", expected, ns)
+		}
+	}
+
+	cfg := &op.Config{}
+
+	manifest, err := mf.NewManifest(resourceWithAnnotation, true, nil)
+	if err != nil {
+		t.Errorf("NewManifest() = %v, wanted no error", err)
+	}
+
+	if err := transformManifest(cfg, &manifest); err != nil {
```

Comment:
how about this: instead of using `transformManifest` from the controller.

shall we create a separate list 

```
fs := []mf.Transformer{
		transform.InjectNamespaceConditional(flag.AnnotationPreserveNS, <new namespace>),
	}
manifest.Transform(tfs...)
```

Reply:
Done